main.go

```diff
@@ -14,50 +14,21 @@ import (
 	"github.com/antonlindstrom/mesos_stats"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/log"
+	"github.com/pvdissel/mesos_metrics"
 )
 
 const concurrentFetch = 100
 
 // Commandline flags.
 var (
-	addr           = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
-	autoDiscover   = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
-	localURL       = flag.String("exporter.local-url", "http://127.0.0.1:5051", "URL to the local Mesos slave")
-	masterURL      = flag.String("exporter.discovery.master-url", "http://mesos-master.example.com:5050", "Mesos master URL")
-	metricsPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics")
+	addr = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
+	autoDiscover = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
+	localURL = flag.String("exporter.local-url", "http://127.0.0.1:5051", "URL to the local Mesos slave")
+	masterURL = flag.String("exporter.discovery.master-url", "http://mesos-master.example.com:5050", "Mesos master URL")
```

Comment:
Masters are always a bit messy due to being multiple of them that are (in-theory) in sync. If it's a situation where there's an elected leader that is authoritative for long periods of time (hours) and handles all traffic then you have the option of what you're doing here - but if the other masters do anything that's worth monitoring then you also need to scrape them.

What you end you usually doing is have some exporter sitting beside each master that scrapes only to that master. Depending on whether the master is leader or follower it may return different metrics, and then you handle all that on the prometheus end (via averaging of some form usually and/or only aggregating up stats from masters identifying themselves as the leader).

Slaves are simpler, you run an exporter beside each that prometheus scrapes. That takes care of the scrape timing and timestamps.